core/lib/resources/presence/presence_manager.js

```diff
@@ -118,8 +126,14 @@ PresenceManager.prototype.stampExpiration = function(message) {
   message.at = Date.now() + PresenceManager.messageExpiry;
 };
 
+// FIXME: Method signature is getting unmanageable.  
 PresenceManager.prototype.addClient = function(socketId, userId, userType,
-                                                      userData, callback) {
+                                                      userData, clientData, callback) {
+
+  if (typeof(clientData) === 'function') {
+    callback = clientData;
+  }
```

Comment:
Right tests. Then let's keep this.

I don't think we should care about anyone using this directly. These are internal apis and there is no guarantees about how the internal apis change.